synapse/api/auth/msc3861_delegated.py

```diff
@@ -243,6 +243,24 @@ async def get_user_by_req(
             # so that we don't provision the user if they don't have enough permission:
             requester = await self.get_user_by_access_token(access_token, allow_expired)
 
+            # Allow impersonation by an admin user using `user_id` query parameter
+            if request.args is not None:
+                user_id_params = request.args.get(b"user_id")
+                if user_id_params:
+                    if SCOPE_SYNAPSE_ADMIN in requester.scope:
+                        user_id_str = user_id_params[0].decode("ascii")
+                        impersonated_user_id = UserID.from_string(user_id_str)
+                        logging.info(f"Admin impersonation of user {user_id_str}")
+                        requester = create_requester(
+                            user_id=impersonated_user_id,
+                            scope=[SCOPE_MATRIX_API],
```

Comment:
```suggestion
                            scope=[SCOPE_MATRIX_API],
                            authenticated_entity=requester.user,
```

or something? That way we will log that this was done as an impersonation, and ratelimiting does the right thing, etc